src/main/java/rx/internal/operators/OperatorSampleWithObservable.java

```diff
@@ -57,16 +59,18 @@ public void onNext(U t) {
 
             @Override
             public void onError(Throwable e) {
+                // no need to null check, main is assigned before any of the two gets subscribed
+                main.get().unsubscribe();
                 s.onError(e);
-                unsubscribe();
             }
 
             @Override
             public void onCompleted() {
+                // no need to null check, main is assigned before any of the two gets subscribed
+                main.get().unsubscribe();
```

Comment:
The main and child are on different subscription container so child can't use anything from main. But if you think otherwise, please write a unit test.